scripts/dndrks/cranes.lua

```diff
@@ -0,0 +1,393 @@
+-- cranes
+-- one buffer, two voices.
+-- ---------------------
+-- at start:
+-- press key 2 to rec.
+-- then:
+-- press key 2 to play.
+-- ..
+-- listen: it's buffer 1.
+--
+-- ////
+-- head to params to find
+-- speed + vol.
+-- \\\\
+--
+-- *** back home: ***
+-- key 1 (hold) = toggle loop pts
+--    for buffer 2
+-- key 2 (after loop playing) =
+--    toggle overwrite
+-- key 3 = ~~ / 0.5 / 1.5 / 2
+--    (see params)
+-- key 3 + key 1 (hold) = erase
+-- enc 1 = overwrite (see key 2)
+--    (0 = add, 1 = clear)
+-- enc 2/3 = buff 1 + 2 loop pts
+--
+
+engine.name = "SoftCut"
+
+-- counting ms between key 2 taps
+-- sets loop length
+function count()
+  rec_time = rec_time + 0.01
+end
+
+-- track recording state
+rec = 0
+
+function init()
+
+  engine.loop_start(1,0)
+  engine.loop_end(1,60)
+  engine.loop_on(1,1)
+  engine.pre(1,1)
+  engine.amp(1,1)
+  engine.rec_on(1,1)
+  engine.rec(1,1)
+  engine.adc_rec(1,1,1)
+  engine.adc_rec(1,2,1)
+  engine.play_dac(1,1,1)
+  engine.play_dac(1,2,1)
+  engine.rate(1,1)
+  engine.rate_lag(1,0.6)
+  engine.reset(1)
+
+  engine.play_rec(1, 2, 1)
+  engine.play_dac(2, 1, 1)
+  engine.play_dac(2, 2, 1)
+  engine.rec(2, 0)
+  engine.pre(2, 1)
+  engine.amp(2, 1)
+  engine.rec_on(2, 1)
+  engine.loop_on(2, 1)
+  engine.loop_start(2, 0)
+  engine.loop_end(2, 60)
+  engine.reset(2)
+  engine.rate(2,1)
+  engine.rate_lag(2,0.35)
+
+  engine.clear()
+  engine.stop(1)
+  engine.stop(2)
+
+  params:add_option("speed buffer 1", speedlist)
```

Comment:
hi @artfwo -- thank you so much!

I tried updating these initializers, but got the following errors upon run:

```
### SCRIPT ERROR: init
/home/we/norns/lua/params/option.lua:16: bad argument #1 to 'pairs' (table expected, got string)
stack traceback:
	/home/we/norns/lua/norns.lua:159: in function </home/we/norns/lua/norns.lua:159>
	[C]: in function 'pairs'
	/home/we/norns/lua/params/option.lua:16: in function 'params/option.new'
	/home/we/norns/lua/paramset.lua:53: in function 'paramset.add_option'
	/home/we/dust/scripts/derks/cranes_oct_2018.lua:75: in function 'init'
	[C]: in function 'xpcall'
	/home/we/norns/lua/norns.lua:160: in field 'try'
	/home/we/norns/lua/script.lua:52: in function 'script.init'
	/home/we/norns/lua/script.lua:91: in function </home/we/norns/lua/script.lua:90>
	[C]: in function 'xpcall'
	/home/we/norns/lua/norns.lua:160: in field 'try'
	/home/we/norns/lua/engine.lua:87: in function </home/we/norns/lua/engine.lua:85>
```

I have 180904 installed. any assistance or further documentation about this formatting would be appreciated! <3

Reply:
hey dan! excited to try cranes!

you submitted right at a moment where we're switching syntax and about to release an update (either today or tomorrow).

you can immediately get up-to-date by doing a `git pull` inside your `norns` folder (not sure if you're using git on the norns itself... if not i can provide more instruction).

in the meantime i can test this out and bundle it into the update!

Reply:
shweeeee! stoked!

I'm not using git on my norns -- quick poke around a few threads on lines didn't surface instruction. happy to jump in if you can assist with direction? tytytytyty

Reply:
connect via wifi router (not hotspot).
open a terminal. `ssh we@ip` where ip is the ip address of the norns (shown in system menu)
then
```
cd norns
./stop.sh
git pull
./waf
sudo reboot now
```

this will get your software updated. you'll still be able to install the new full update when i post it

Reply:
actually--- this is a lot of extra work for you. i'm going to just merge this and fix the syntax, then it'll be available with the update and you can just run a normal update.

Reply:
OH! cool! alright, well, this is good for next time for sure. thanks so much Brian and Artem!